device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -466,41 +462,13 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
                      in.begin() + n_candidates_per_step[it], out.begin());
     }
 
-    // Get the number of tips per step
-    std::vector<unsigned int> n_tips_per_step;
-    n_tips_per_step.reserve(n_steps);
-    for (unsigned int it = 0; it < n_steps; it++) {
-        n_tips_per_step.push_back(m_copy.get_size(tips_map[it]));
-    }
-
-    // Copy tips_map into the tips vector (D->D)
-    unsigned int n_tips_total =
-        std::accumulate(n_tips_per_step.begin(), n_tips_per_step.end(), 0);
-    vecmem::data::vector_buffer<typename candidate_link::link_index_type>
-        tips_buffer{n_tips_total, m_mr.main};
-    m_copy.setup(tips_buffer)->ignore();
-
-    vecmem::device_vector<typename candidate_link::link_index_type> tips(
-        tips_buffer);
-
-    unsigned int prefix_sum = 0;
-
-    for (unsigned int it = 0; it < n_steps; it++) {
-        vecmem::device_vector<typename candidate_link::link_index_type> in(
-            tips_map[it]);
-
-        const unsigned int n_tips = n_tips_per_step[it];
-        if (n_tips > 0) {
-            thrust::copy(thrust::cuda::par.on(stream), in.begin(),
-                         in.begin() + n_tips, tips.begin() + prefix_sum);
-            prefix_sum += n_tips;
-        }
-    }
-
     /*****************************************************************
      * Kernel6: Build tracks
      *****************************************************************/
 
+    // Get the number of tips
+    auto n_tips_total = m_copy.get_size(vecmem::get_data(tips_buffer));
```

Comment:
🤔 I would probably skip out on calling `vecmem::get_data` explicitly. It's not technically needed here, and I don't think it improves readability.

```suggestion
    auto n_tips_total = m_copy.get_size(tips_buffer);
```